### PR TITLE
parseDate fixed for h2 and presto.

### PR DESCRIPTION
**Pull request by openceo**
parseDate is a pure function that is mapped to parsedatetime and date_parse in H2 and Presto databases. Fixed this pure function to correctly generate SQL for these DBs,

### CODE REVIEW

**Comment by epsstan** [comment]
Can we improve the tests for this PR ? by running the tests against a Presto instance like we do for SQLServer

**Comment by epsstan** [comment]
@vikask-gs - Can you take a look at this PR please ? Especially as it relates to tests and the your refactoring of the database test framework ?

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure [hunk 1]
```diff
@@ -157,6 +157,14 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::h2::c
       'cast( parseDateTime('+$params->at(0)+','+$dateFormat +') as date)';
 }
 
+function <<access.private>> meta::relational::functions::sqlQueryToString::h2::transformParseDateParams(params:String[*]):String[2]
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure:160]: It seems that you want to make the format an optional parameter. Original behavior remains same. Correct? What is the purpose of this change?

**Comment by finos-admin** [comment]
This PR is stale because it has been open for 30 days with no activity. Please remove stale label or add any comment to keep this open. Otherwise this will be closed in 5 days.

**Comment by finos-admin** [comment]
This PR was closed because it has been inactive for 35 days. Please re-open if this PR is still relevant.
